### DISCUSSION TITLE
When is Github Classroom making the switch to forked assignments?

### DISCUSSION THREAD

**Discussion by w0244079** [community/community#80909] [GitHub Education] [answered]
### Hi everyone,

Hi there!

Just wondering if we know when the switch will be made to having students fork assignments rather than clone templates. Is there a date set?

Thanks!

**Comment by Bardin08** [comment]
Hi, @w0244079. There's no announced date yet. The only thing we know is that it will be in January.

https://github.blog/changelog/2023-12-08-upcoming-changes-to-assignment-repositories-and-starter-code-in-github-classroom/

**Comment by daisla** [comment]
I would also love an answer to this. The blog post says "in January" ... well we're approaching mid-January, and the old system seems to still be in effect. Now spring classes are getting going, and I would love not to kick things off with something that is about to be deprecated.
(Fwiw, moving to forks is a great move -- going to be way better for our purposes.)

**Reply by RyanHecht** [reply]
We won't deprecate the old version in the middle of the semester! The Public Beta will begin this month, and we're currently targeting mid-to-late of June for the transition from Public Beta to General Availability.

Glad to hear forks is better for your use case! We can't wait to begin the beta :)

**Reply by RyanHecht** [reply]
Hey @daisla, you may be interested in knowing that the beta has begun! https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/

**Comment by erdogmush** [comment]
We really need a response to this one! I had to set up classrooms for the Spring semester (the old system is still in place), and if the old system will be deprecated in the middle of the semester, I don't know what effect this would have on the classroom-created repos that already have student-added content in them by the deprecation date. Also hug thumbs up for the new fork-based system! Great idea and would make deploying patches so much easier!

**Reply by RyanHecht** [reply]
We won't deprecate the old version in the middle of the semester! The Public Beta will begin this month, and we're currently targeting mid-to-late of June for the transition from Public Beta to General Availability 

I'm glad to hear you're excited for the fork-based system! We can't wait to see what new workflows this opens up for our teachers :)

**Reply by RyanHecht** [reply]
Hey @erdogmush, just letting you know that the beta has begun! https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/

**Comment by w0244079** [comment]
Hi Everyone…..my understanding is that when it rolls out it will be a beta program that you must opt into which means that in theory it should not disrupt your delivery for this semester if you choose to not switch over. I could be wrong but I read this recently. Anyone at GH able to confirm this?

From: Hakan Erdogmus ***@***.***>
Date: Saturday, January 13, 2024 at 1:39 PM
To: community/community ***@***.***>
Cc: Caines,Michael ***@***.***>, Mention ***@***.***>
Subject: Re: [community/community] When is Github Classroom making the switch to forked assignments? (Discussion #80909)

CAUTION: This message was sent from outside the organization. Please do not click links or open attachments unless you recognize the source of this email and know the content is safe.
________________________________

We really need a response to this one! I had to set up classrooms for the Spring semester (the old system is still in place), and if the old system will be deprecated in the middle of the semester, I don't know what effect this would have on the classroom-created repos that already have student-added content in them by the deprecation date. Also hug thumbs up for the new fork-based system! Great idea and would make deploying patches so much easier!

**Reply by mishmanners** [reply]
Yes the public beta is scheduled to start in January as per the blog post. It's still January with a couple more weeks to go 😉

**Reply by RyanHecht** [reply]
This is correct, the opt-in Public Beta should not disrupt your delivery for this semester! After we begin the Public Beta this month, we're targeting a transition to General Availability in mid-to-late June!

**Comment by ron-reaser** [comment]
I also need to know when this change is occurring. I need to release my term project repository by Friday 1/19. I only get one shot for the entire spring semester, and I would prefer to do so under the new system to avoid merge conflicts.

If it's an opt-in beta, I certainly want to opt-in before my release date.

**Reply by michelledion** [reply]
FWIW, I've been able to use [these instructions](https://github.com/konzy/mass_clone) to push updates to student repos after they have been created in the current setup. Not as easy as it will be with the change that's coming, but you _can_ push updates, if you need to in the current setup.

**Reply by ron-reaser** [reply]
Thanks. I actually don't need that, since I just teach my students to add the upstream repo, fetch, and merge. The issue I have with the template-based it approach is how the upstream repo's commit history is squashed into a new commit when generating the student repos. It causes the history of their repo and mine to diverge, leading to merge conflicts later in the course. It also doesn't teach them how to interact with the most common workflow on GitHub, which is based on forking. So that's why I'm hoping to opt in this semester before starting. I hope it won't be too late.

**Reply by RyanHecht** [reply]
The beta has begun! https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/

**Comment by RyanHecht** [comment, marked as answer]
Hey folks! PM of Classroom here. We're planning on rolling this out as a public beta very soon (there are a few last-minute bugs we're sorting out!). As some have pointed out, you will be able to _opt in_ to this new behavior on a per-Classroom basis when the beta begins, so this won't affect any classes in progress! All Classrooms will be transitioned to the new approach when we go from Public Beta to General Availability, which we expect to do after the spring term has ended for the lion's share of our users (currently we're targeting mid-to-late June). When we eventually transition all Classrooms to the forks-based approach, previously accepted assignments will still work fine; only _newly accepted_ student assignments will be subject to the "important changes" lined out in last month's [Changelog Post](https://github.blog/changelog/2023-12-08-upcoming-changes-to-assignment-repositories-and-starter-code-in-github-classroom/).

We're looking forward to starting the beta of our forks-based approach very soon! :)



EDIT 1/23: The beta has begun! Learn more here: https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/

**Reply by jazzij** [reply]
How can we opt to be notified of the beta release?

**Reply by halfond** [reply]
Thanks for the additional information about this will roll out, appreciate knowing this will happen in a way that won't disrupt most classes.

**Reply by RyanHecht** [reply]
The beta is now released, @jazzij! https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/
